core/src/main/java/com/linecorp/armeria/client/ConnectionPoolListener.java

```diff
@@ -122,4 +124,19 @@ void connectionClosed(SessionProtocol protocol,
     default ConnectionPoolListener unwrap() {
         return this;
     }
+
+    default ConnectionPoolListener doThen(ConnectionPoolListener anotherConnectionPoolListener) {
+        requireNonNull(anotherConnectionPoolListener, "anotherConnectionPoolListener");
+        return new ConnectionPoolListener() {
+            @Override
+            public void connectionOpen(SessionProtocol protocol, InetSocketAddress remoteAddr, InetSocketAddress localAddr, AttributeMap attrs) throws Exception {
+                anotherConnectionPoolListener.connectionOpen(protocol, remoteAddr, localAddr, attrs);
```

Comment:
I think the expectation is that `this#connectionOpen` is called, and then `anotherConnectionPoolListener#connectionOpen` is called (hence "chained").

Can you also add a test to confirm this behavior?

```suggestion
                connectionOpen(protocol, remoteAddr, localAddr, attrs);
                anotherConnectionPoolListener.connectionOpen(protocol, remoteAddr, localAddr, attrs);
```